internal/grpc/services/publicstorageprovider/publicstorageprovider.go

```diff
@@ -0,0 +1,315 @@
+package publicstorageprovider
+
+import (
+	"context"
+	"fmt"
+	"path"
+	"strings"
+
+	link "github.com/cs3org/go-cs3apis/cs3/sharing/link/v1beta1"
+	provider "github.com/cs3org/go-cs3apis/cs3/storage/provider/v1beta1"
+	"github.com/cs3org/reva/pkg/rgrpc"
+	"github.com/mitchellh/mapstructure"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+	"go.opencensus.io/trace"
+	"google.golang.org/grpc"
+	codes "google.golang.org/grpc/codes"
+	gstatus "google.golang.org/grpc/status"
+)
+
+func init() {
+	rgrpc.Register("publicstorageprovider", New)
+}
+
+type config struct {
+	MountPath               string `mapstructure:"mount_path"`
+	MountID                 string `mapstructure:"mount_id"`
+	PublicShareProviderAddr string `mapstructure:"public_share_provider_addr"`
+	StorageProviderAddr     string `mapstructure:"storage_provider_addr"`
+}
+
+type service struct {
+	conf                 *config
+	mountPath, mountID   string
+	publicShareProviderC link.LinkAPIClient
+	storageProviderC     provider.ProviderAPIClient
```

Comment:
Use the gateway instead, services should not connect directly to storage provider.